src/sail_lean_backend/pretty_print_lean.ml

```diff
@@ -238,14 +238,37 @@ let rec captured_typ_var ((i, Typ_aux (t, _)) as typ) =
 
 let doc_typ_id ctx (typ, fid) = flow (break 1) [doc_id_ctor fid; colon; doc_typ ctx typ]
 
-let doc_kind (K_aux (k, _)) = match k with K_int -> string "Int" | K_bool -> string "Bool" | K_type -> string "Type"
+let doc_kind ctx (kid : kid) (K_aux (k, _)) =
+  match k with
+  | K_int -> if provably_nneg ctx (Nexp_aux (Nexp_var kid, Unknown)) then string "Nat" else string "Int"
+  | K_bool -> string "Bool"
+  | K_type -> string "Type"
 
-let doc_quant_item ctx (QI_aux (qi, _)) =
+let doc_quant_item_all ctx (QI_aux (qi, _)) =
   match qi with
-  | QI_id (KOpt_aux (KOpt_kind (k, ki), _)) -> flow (break 1) [doc_kid ctx ki; colon; doc_kind k]
+  | QI_id (KOpt_aux (KOpt_kind (k, ki), _)) -> flow (break 1) [doc_kid ctx ki; colon; doc_kind ctx ki k]
   | QI_constraint c -> doc_nconstraint ctx c
 
-let doc_typ_quant ctx tq = match tq with TypQ_tq qs -> List.map (doc_quant_item ctx) qs | TypQ_no_forall -> []
+(* Used to annotate types with the original constraints *)
+let doc_typ_quant_all ctx tq = match tq with TypQ_tq qs -> List.map (doc_quant_item_all ctx) qs | TypQ_no_forall -> []
+
+let doc_quant_item_relevant ctx (QI_aux (qi, annot)) =
+  match qi with
+  | QI_id (KOpt_aux (KOpt_kind (k, ki), _)) -> Some (flow (break 1) [doc_kid ctx ki; colon; doc_kind ctx ki k])
+  | QI_constraint c -> None
+
+let gather_constraints ctx tq =
```

Comment:
`gather_constraints` appears to be unused.  Subsumed by `Env.add_typquant`?

Reply:
Ah, good catch!